etl/steps/data/garden/wid/2023-01-27/issues.py

```diff
@@ -0,0 +1,173 @@
+# %% [markdown]
+# # Checks to WID data
+# The data here is harmonized to the country and year format Our World in Data uses
+
+import pandas as pd
+
+# %%
+from owid.catalog import Dataset
+
+from etl.paths import DATA_DIR
+
+# # To show the entire output of the dataframess
+# pd.set_option("display.max_rows", None)
+
+# Variable type suffixes
+vars = ["pretax", "posttax_dis", "posttax_nat", "wealth"]
+
+# Decile prefixes to evaluate
+deciles = ["p0p10", "p10p20", "p20p30", "p30p40", "p40p50", "p50p60", "p60p70", "p70p80", "p80p90", "p90p100"]
+
+# Metrics to evaluate
+metric = ["avg", "thr", "share"]
+
+ds = Dataset(DATA_DIR / "garden" / "wid" / "2023-01-27" / "world_inequality_database")
+print(ds.table_names)
+
+df = ds["world_inequality_database"]
+df = pd.DataFrame(df)
+
+# %% [markdown]
+# ## Check Gini values
+
+# %%
+# Check Gini values
+# For each type of income/wealth
+for v in vars:
+    # Define column
+    col = f"p0p100_gini_{v}"
+    # Filter only ginis lower than 0 or higher than 1
+    df_review = df[(df[col] > 1) | (df[col] < 0)]
+
+    # Print no issues if no values have been found
+    if len(df_review) == 0:
+        print(f"No issues for {v}")
+
+    # Print the observations with issues if values are found
+    else:
+        print(f"There are {len(df_review)} observations with issues in {v}:")
+        print(df_review[[col]])
+
+# %% [markdown]
+# ## Check sum of shares
+
+# %%
+# CHECK SUM OF SHARES
+
+# For each income/wealth variable
+for v in vars:
+    # Set columns to evaluate
+    cols = [f"{s}_share_{v}" for s in deciles]
+    # Get sum of shares
+    df["sum"] = df[cols].sum(1)
+    # Count the nulls between the 10 decile share variables
+    df["null_check"] = df[cols].isnull().sum(1)
+
+    # Exclude sum of shares = 0
+    df_review = df[df["sum"] != 0]
+    # Only keep shares with no null values
+    df_review = df_review[df_review["null_check"] == 0]
+    # Look for unusual sum values
+    df_review = df_review[(df_review["sum"] > 100.2) | (df_review["sum"] < 99.8)]
+
+    # Print no issues if nothing is found
+    if len(df_review) == 0:
+        print(f"No issues for {v}")
+
+    # Print the observations with issues otherwise
+    else:
+        print(f"There are {len(df_review)} observations with issues in {v}:")
+        print(df_review["sum"])
```

Comment:
General advice (maybe not for all cases): If you want to check things, it may be better to use assertions, like `assert variable >0, "Unexpected value for variable"`. Otherwise, you will need to read through lots of prints, and that will simply take you more time. So, print only if something is unexpected, otherwise don't print anything.